File: python_fancy/your_bar.py
#!/usr/bin/python3

import sys

def start_it():
    print("\n  ***            Percent Complete                ***")
    print("  |                                                |")
    print("  0% ################## 50% ################### 100%")
    print("  |                                                |")
    sys.stdout.write("  ")
    sys.stdout.flush()

def add_star():
    sys.stdout.write("*")
    sys.stdout.flush()

def end_it():
    sys.stdout.write("\n\n")

File: ruby_fancy/YourBar.rb
#!/usr/bin/ruby

class YourBar
    def self.start_it()
        print "\n  ***            Percent Complete                ***\n"
        print "  |                                                |\n"
        print "  0% ################## 50% ################### 100%\n"
        print "  |                                                |\n"
        print "  "
    end

    def self.add_star()
        print "*"
    end

    def self.end_it()
        print "\n\n"
    end
end
